### PR TITLE
feat(forge) add projectRoot cheat code

### PR DESCRIPTION
**Pull request by jdgc**
## Motivation

Hi, first time contributor so please let me know if anything is off.

This PR closes https://github.com/foundry-rs/foundry/issues/2719

### CODE REVIEW

File: evm/src/executor/inspector/cheatcodes/ext.rs [hunk 1]
```diff
@@ -189,6 +189,15 @@ fn get_env(key: &str, r#type: ParamType, delim: Option<&str>) -> Result<Bytes, B
     value_to_abi(val, r#type, is_array)
 }
 
+fn project_root(state: &Cheatcodes) -> Result<Bytes, Bytes> {
+    let root =
+        state.config.root.clone().into_os_string().into_string().map_err(|_| {
+            util::encode_error("invalid characters in project root path".to_string())
+        })?;
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/ext.rs:196]: we should just assume valid paths and simply do `let root = format!("{}", state.config.root.display())`

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, this looks good, 

smol nit re path to string

File: evm/src/executor/inspector/cheatcodes/ext.rs (hunk 1 above)
Comment (jdgc) [reply evm/src/executor/inspector/cheatcodes/ext.rs]: sounds good, updated in d235f827b604642286fca34fe64e7c99786f8826

**Review by mattsse — APPROVED** [review]
thanks.

could you please also document this in the book https://github.com/foundry-rs/book/issues/523

**Pull request merged**